internal/mtail/mtail.go

```diff
@@ -58,9 +58,10 @@ type Server struct {
 	dumpAstTypes bool // if set, mtail prints the program syntax tree after type checking
 	dumpBytecode bool // if set, mtail prints the program bytecode after code generation
 
-	syslogUseCurrentYear bool // if set, use the current year for timestamps that have no year information
-	omitMetricSource     bool // if set, do not link the source program to a metric
-	omitProgLabel        bool // if set, do not put the program name in the metric labels
+	syslogUseCurrentYear    bool           // if set, use the current year for timestamps that have no year information
+	omitMetricSource        bool           // if set, do not link the source program to a metric
+	omitProgLabel           bool           // if set, do not put the program name in the metric labels
+	storeExpireTickInterval *time.Duration // interval to expire/delete metrics from store
```

Comment:
You could not have a pointer here and instead leave it as the zero value, and then we can change my suggested `if != nil` to if duration > 0 and put it inside StartExpiryLoop() instead.